src/background/hypothesis-chrome-extension.js

```diff
@@ -178,14 +183,24 @@ export default function HypothesisChromeExtension({
   // Two important events are when the tab's `status` changes to `loading`
   // when the user begins a new navigation and when the tab's status changes
   // to `complete` after the user completes a navigation
-  function onTabUpdated(tabId, changeInfo, tab) {
-    if (changeInfo.status === TAB_STATUS_LOADING) {
-      resetTabState(tabId, tab.url);
-      const query = directLinkQuery(tab.url);
+  /**
+   *
+   * @param {number} tabId
+   * @param {chrome.tabs.TabChangeInfo} changeInfo
+   * @param {chrome.tabs.Tab} tab
+   */
+  function onTabUpdated(tabId, { status }, tab) {
```

Comment:
Reading the documentation for the `changeInfo` parameter (see https://developer.chrome.com/extensions/tabs#event-onUpdated) it mentions that the `changeInfo` object only includes a `url` field if the URL has changed. Does this mean that we could potentially get Chrome to track whether the URL has changed rather than requiring us to track it ourselves?

Reply:
Unfortunately, that doesn't work for page reloads when the `changeInfo.url` is always `undefined`.